src/main/java/org/apache/parquet/hadoop/BdecParquetWriter.java

```diff
@@ -131,6 +131,10 @@ public List<Long> getRowCountsFromFooter() {
     return blockRowCounts;
   }
 
+  public List<BlockMetaData> getBlocksMetadata() {
+    return writer.getFooter().getBlocks();
```

Comment:
can getFooter be null?
can getBLocks be null?

If either of these is possible please return an Optional<List<BlockMetaData>> instead of expecting the caller to know that the value can be null

Reply:
Yes. Resolved as we no longer need this anymore.